pkg/transform/transform.go

```diff
@@ -0,0 +1,158 @@
+package transform
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/fusor/cpma/env"
+	"github.com/fusor/cpma/pkg/io"
+	"github.com/fusor/cpma/pkg/transform/oauth"
+	"github.com/fusor/cpma/pkg/transform/secrets"
+	"github.com/sirupsen/logrus"
+)
+
+// OCP4InstallMsg message about using generated manifests
+const OCP4InstallMsg = `To install OCP4 run the installer as follow in order to add CRDs:
+' /openshift-install --dir $INSTALL_DIR create install-config'
+'./openshift-install --dir $INSTALL_DIR create manifests'
+# Copy generated CRD manifest files  to '$INSTALL_DIR/openshift/'
+# Edit them if needed, then run installation:
+'./openshift-install --dir $INSTALL_DIR  create cluster'`
+const MasterConfigFile = "/etc/origin/master/master-config.yaml"
+const NodeConfigFile = "/etc/origin/node/node-config.yaml"
+const RegistriesConfigFile = "/etc/containers/registries.conf"
+
+type Cluster struct {
+	Master Master
+}
+
+type Master struct {
+	OAuth   oauth.OAuthCRD
+	Secrets []secrets.Secret
+}
+
+type Manifest struct {
+	Name string
+	CRD  []byte
+}
+
+type Config struct {
+	MasterConfigFile     string
+	NodeConfigFile       string
+	RegistriesConfigFile string
+	OutputDir            string
+	Hostname             string
+}
+
+type ManifestTransformOutput struct {
+	//Config    Config
+	Manifests []Manifest
+}
+
+type TransformRunner struct {
+	Config string
+}
+
+// GetFile allows to mock file retrieval
+var GetFile = io.GetFile
+
+func Start() {
+	config := Config{}
+	config.OutputDir = env.Config().GetString("OutputDir")
+	config.Hostname = env.Config().GetString("Source")
+	config.MasterConfigFile = MasterConfigFile
+	config.RegistriesConfigFile = RegistriesConfigFile
+	transformRunner := NewTransformRunner(config)
+
+	if err := transformRunner.Transform([]Transform{
+		OAuthTransform{
+			Config: &config,
+		},
+		SDNTransform{
+			Config: &config,
+		},
+		RegistriesTransform{
+			Config: &config,
+		},
+	}); err != nil {
+		logrus.WithError(err).Fatalf("%s", err.Error())
+	}
+}
+
+// DumpManifests creates OCDs files
+func DumpManifests(manifests []Manifest) {
```

Comment:
[BLOCKER] This is tightly coupled with the semantic purpose of a `ManifestTransformOutput`, so I would move this method into the same file you put that struct and its interface implementations.

Reply:
I moved ManifestTransformOutput stuff to its own file. I think this is resolved.